Usability fixes and improvements when using BUILD_STRATEGY_USE_HEX_FILE (#12543)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Bootloader libraries
--------------------

* :ref:`mcuboot_ncs`:

>   * Allow the path in :kconfig:option:`CONFIG_BOOT_SIGNATURE_KEY_FILE` to be relative to the current sample directory.

Reviewer: ```suggestion
  * The path set with :kconfig:option:`CONFIG_BOOT_SIGNATURE_KEY_FILE` can now be relative to the current sample directory.
```

    See the help text of the kconfig option for more details.

* :ref:`bootloader`:

>   * Fixed: When both :ref:`bootloader` and :ref:`mcuboot_ncs` are present, and MCUboot is provided as a hex file (via :kconfig:option:`CONFIG_MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE`), the provision page was not regenerated correctly.
>     Fix generation of the provision page, and also add the option to provide the provision page as a hex file or, be skipped (:kconfig:option:`CONFIG_PROVISION_BUILD_STRATEGY_USE_HEX_FILE` and :kconfig:option:`CONFIG_PROVISION_BUILD_STRATEGY_SKIP_BUILD`)

Reviewer: ```suggestion
  * Added:

    * the :kconfig:option:`CONFIG_PROVISION_BUILD_STRATEGY_USE_HEX_FILE` Kconfig option to provide the provision page as a HEX file.
    * the :kconfig:option:`CONFIG_PROVISION_BUILD_STRATEGY_SKIP_BUILD` kconfig option to skip the building of the provision page. 
 
  * Fixed:

    * The generation of the provision page.
    * The regeneration of the provision page when both :ref:`bootloader` and :ref:`mcuboot_ncs` are present, and MCUboot is provided as a HEX file (using :kconfig:option:`CONFIG_MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE`).
```

This is a guess, let me know if the descriptions of the options mentioned are correct.
